src/renderer/missionWindow/MissionWindow.tsx

```diff
@@ -447,7 +506,10 @@ export default class MissionWindow extends Component<ThemeProps, State> {
   private unlockParameterInputs(waypointType: string): void {
     const { locked: newLocked } = this.state;
 
-    if (waypointType === 'geofence') {
+    if (waypointType === 'geofenceContainerKeepIn') {
+      newLocked.geofence = false;
+    }
+    if (waypointType === 'geofenceContainerKeepOut') {
```

Comment:
so if I remember right this is the check to disable the lock for the geofencing box. Since you have two geofencing boxes, you will need two locks (one lock for the keep in box, one for the keep out box).

Lets call these locks `geofenceKeepIn` and `geofenceKeepOut`..

Modify the following to accommodate for the locks:
- `Locked`
- `lockedCache`
- this function, you can literally follow the one from [VTOLSearch](https://github.com/NGCP/GCS/blob/master/src/renderer/missionWindow/parameter/VTOLSearch.tsx#L205)
- the `input` fields for Geofencing. the  disabled fields should be modified for the new locks
- the `CreateBoundingBox` button. my naming schema is kinda crappy tbh but you should have the `Geofence Keep-Out` under value and `geofenceKeepIn` or `geofenceKeepOut` under name. See [this line](https://github.com/NGCP/GCS/blob/master/src/renderer/missionWindow/parameter/VTOLSearch.tsx#L236) for how VTOLSearch bounding box does it